engine/include/engine/engine.h

```diff
@@ -8,12 +8,34 @@
 struct ResultSet {
 	std::vector<gdf_column *> columns;
 	std::vector<std::string> names;
-	ral::frame::BlazingTableView *blazingTableView;
+	std::unique_ptr<ral::frame::BlazingTable> blazingTable;
+
+	ResultSet() {
+		columns = {};
+		names = {};
+		blazingTable = nullptr;
+	}
+
+	ResultSet(std::unique_ptr<ral::frame::BlazingTable> blazingTableValue)
+		: blazingTable{std::forward<std::unique_ptr<ral::frame::BlazingTable>>(blazingTableValue)} {}
+
+	ResultSet(const ResultSet & other) {
```

Comment:
this is a copy constructor. If you are trying to define a move constructor then you should use ```ResultSet(const ResultSet && other)```

Reply:
I'm adding the copy constructor beacuse it's a restriction of cython. Internally, cython generates cpp code that uses copy constructor and assign operator.